### ISSUE TITLE
`pyenv install 3.10.0` fails with `Makefile:222: *** missing separator`

### ISSUE THREAD

**Issue by georgevreilly** [pyenv/pyenv#2102]
### Description
- [ ] Platform information (e.g. Ubuntu Linux 16.04): macOS Big Sur 11.6
- [ ] OS architecture (e.g. amd64): x86_64
- [ ] pyenv version: 2.1.0
- [ ] Python version: 3.9.7
- [ ] C Compiler information (e.g. gcc 7.3): clang-1205.0.22.11
- [ ] Please attach verbose build log as gist
  * If the problem happens in a Pyenv invocation, you can turn on debug logging by setting `PYENV_DEBUG=1`, e.g. `env PYENV_DEBUG=1 pyenv install -v 3.6.4`
 

Gist: https://gist.github.com/georgevreilly/1ed2d761b705d2b9808e35309d3e6a05

Looks like there are some embedded newlines in `CONFIG_ARGS`

```
   220	# configure script arguments
   221	CONFIG_ARGS=	 '--prefix=/Users/georgevreilly/.pyenv/versions/3.10.0' '--libdir=/Users/georgevreilly/.pyenv/versions/3.10.0/lib' '--with-openssl=/usr/local/opt/openssl@1.1' '--with-tcltk-libs=-L/usr/local/opt/tcl-tk/lib -ltcl8.6.11_1
   222	8.6.11_1 -ltk8.6.11_1
   223	8.6.11_1' '--with-tcltk-includes=-I/usr/local/opt/tcl-tk/include' 'CC=clang' 'CFLAGS=-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include  ' 'LDFLAGS=-L/usr/local/opt/readline/lib -L/usr/local/opt/readline/lib -L/Users/georgevreilly/.pyenv/versions/3.10.0/lib  -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib' 'CPPFLAGS=-I/usr/local/opt/readline/include -I/usr/local/opt/readline/include -I/Users/georgevreilly/.pyenv/versions/3.10.0/include '
```

Someone else is hitting the same problem: https://stackoverflow.com/questions/69496719/pyenv-install-of-python-errors-with-missing-separators

**Comment by kbreit** [comment]
@georgevreilly Any progress on this? It works on my work laptop but not personal system. They're obviously not identical but relatively similar.

**Comment by native-api** [comment]
@georgevreilly what does `brew --cellar tcl-tk` output for you? Looks like you have multiple versions of Tcl/Tk installed somehow.

**Comment by native-api** [comment]
@kbreit that "8.6.11_1.reinstall" shouldn't be there. Probably a leftover from an unsuccessful package reinstallation.

In any case, we should use a better discovery mechanism that's immune to these effects -- e.g. `brew --prefix tcl-tk`.

**Comment by georgevreilly** [comment]
I have identical results to @kbreit. I don't recall ever installing tcl-tk explicitly, so I assume it's a dependency of something else.

```
> brew --cellar tcl-tk
/usr/local/Cellar/tcl-tk

> ls -l "$(brew --cellar tcl-tk)"
total 0
drwxr-xr-x  12 georgevreilly  staff  384 Aug 30 21:55 8.6.11_1
drwxr-xr-x  12 georgevreilly  staff  384 Feb 12  2021 8.6.11_1.reinstall
```

**Comment by kbreit** [comment]
This fixed the problem for me too. Thank you!

**Comment by JoeGermuska** [comment]
I ran into this trying to install python 3.9.7. Tracking down the Make errors, I found that on two lines, newlines had been inserted where they didn't belong. Simply by manually rejoining the split lines, I was able to manually run `make` and `make install` and it now shows up in `pyenv versions`

tl;dr: Is pyenv causing newlines to be created in Makefiles? If so, this should be reopened, or a new issue created. I haven't spent the time to figure out how the Makefiles are generated.
